weatherService.js

```diff
@@ -0,0 +1,46 @@
+const geoCode = require('./util/geoCode')
+const forecast = require('./util/forecast')
+const apiError = require('./error/ApiError')
+
+//TODO: improvise the code
+const weatherAPI = async (req) => {
+    try {
+        const response = [];
+        const geoResponse = await geoCode(req);
+        const data = geoResponse.data;
+        await Promise.all(
+            data.features.map(async (test) => {
+
+                const [latitude, longitude] = test.center;
+                const place = test.place_name;
+                const {data} =  await forecast(latitude, longitude);
+                const currentData = data.current;
+
+                if(currentData !== undefined) {
+                    response.push({
```

Comment:
I don't think we need to do push to response. You can just do `const response = await Promise.all()`